src/api-binder.coffee

```diff
@@ -198,6 +198,20 @@ module.exports = class APIBinder
 				@config.set(configToUpdate)
 			.then =>
 				@eventTracker.track('Device bootstrap success')
+		# Check if we need to pin the device, regardless of if we provisioned
```

Comment:
For this to actually happen regardless of if we provisioned, the check in `provisonDevice` below (line 233) should be `if !conf.provisioned or conf.apiKey? or conf.pinDevice?` (and adding pinDevice to the config.getMany there)